sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -276,25 +280,30 @@ export abstract class V4PositionManager {
 
     // If migrating, we need to settle and sweep both currencies individually
     if (isMint(options) && options.migrate) {
-      // payer is v4 positiion manager
+      if (options.useNative) {
+        planner.addUnwrap(BigNumber.from('0x8000000000000000000000000000000000000000000000000000000000000000'))
+      }
+      // payer is v4 position manager
       planner.addSettle(position.pool.currency0, false)
       planner.addSettle(position.pool.currency1, false)
       planner.addSweep(position.pool.currency0, options.recipient)
       planner.addSweep(position.pool.currency1, options.recipient)
+      // sweep wrapped native for good measure
+      if (options.useNative) {
+        planner.addSweep(position.pool.currency0.wrapped, options.recipient)
+      }
     } else {
       // need to settle both currencies when minting / adding liquidity (user is the payer)
       planner.addSettlePair(position.pool.currency0, position.pool.currency1)
     }
 
     // Any sweeping must happen after the settling.
     let value: string = toHex(0)
-    if (options.useNative) {
-      invariant(position.pool.currency0.isNative || position.pool.currency1.isNative, NO_NATIVE)
-      let nativeCurrency: Currency = position.pool.currency0.isNative
-        ? position.pool.currency0
-        : position.pool.currency1
-      value = position.pool.currency0.isNative ? toHex(amount0Max) : toHex(amount1Max)
-      planner.addSweep(nativeCurrency, MSG_SENDER)
+    // When not migrating and adding native currency, add a final sweep
+    if (options.useNative && !(isMint(options) && options.migrate)) {
```

Comment:
I'm confused on this case? why do we need to add  `!(isMint(options) && options.migrate)`

and why do we take out the invariants?

Reply:
Oh i think we check it above?

Reply:
eek just finding the if else branches hard to reason though gonna think if theres a way to clean it up a tad

Reply:
i took out `|| position.pool.currency1.isNative` because this will never be the case - if one of the currencies is native, it will always be currency0

Reply:
added this: `!(isMint(options) && options.migrate)` because we should only add an extra sweep if we are not migrating. The 2 sweeps are already covered when migrating, so we don't need to add an extra

Reply:
why cant it just be `&& !options.migrate`? Why do you need to check the mint options too?

Reply:
typescript is annoying and says that migrate doesnt exist on addLiquidityOptions since it doesnt exist on IncreaseLiquidityOptions. so u need to make sure its a mint